File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/frontend/provider/lib/src/model/financeiro/fin_extrato_conta_banco.dart
/*
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [FIN_EXTRATO_CONTA_BANCO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import 'dart:convert';

import 'package:intl/intl.dart';
import 'package:fenix/src/infra/biblioteca.dart';
import 'package:fenix/src/model/model.dart';

class FinExtratoContaBanco {
	int id;
	int idBancoContaCaixa;
	String mesAno;
	String mes;
	String ano;
	DateTime dataMovimento;
	DateTime dataBalancete;
	String historico;
	String documento;
	double valor;
	String conciliado;
	String observacao;
	BancoContaCaixa bancoContaCaixa;

	FinExtratoContaBanco({
			this.id,
			this.idBancoContaCaixa,
			this.mesAno,
			this.mes,
			this.ano,
			this.dataMovimento,
			this.dataBalancete,
			this.historico,
			this.documento,
			this.valor,
			this.conciliado,
			this.observacao,
			this.bancoContaCaixa,
		});

	static List<String> campos = <String>[
		'ID', 
		'MES_ANO', 
		'MES', 
		'ANO', 
		'DATA_MOVIMENTO', 
		'DATA_BALANCETE', 
		'HISTORICO', 
		'DOCUMENTO', 
		'VALOR', 
		'CONCILIADO', 
		'OBSERVACAO', 
	];
	
	static List<String> colunas = <String>[
		'Id', 
		'Mês/Ano', 
		'Mês', 
		'Ano', 
		'Data de Movimento', 
		'Data do Balancete', 
		'Histórico', 
		'Documento', 
		'Valor', 
		'Conciliado', 
		'Observação', 
	];

	FinExtratoContaBanco.fromJson(Map<String, dynamic> jsonDados) {
		id = jsonDados['id'];
		idBancoContaCaixa = jsonDados['idBancoContaCaixa'];
		mesAno = jsonDados['mesAno'];
		mes = jsonDados['mes'];
		ano = jsonDados['ano'];
		dataMovimento = jsonDados['dataMovimento'] != null ? DateTime.tryParse(jsonDados['dataMovimento']) : null;
		dataBalancete = jsonDados['dataBalancete'] != null ? DateTime.tryParse(jsonDados['dataBalancete']) : null;
		historico = jsonDados['historico'];
		documento = jsonDados['documento'];
		valor = jsonDados['valor'] != null ? jsonDados['valor'].toDouble() : null;
		conciliado = getConciliado(jsonDados['conciliado']);
		observacao = jsonDados['observacao'];
		bancoContaCaixa = jsonDados['bancoContaCaixa'] == null ? null : new BancoContaCaixa.fromJson(jsonDados['bancoContaCaixa']);
	}

	Map<String, dynamic> get toJson {
		Map<String, dynamic> jsonDados = new Map<String, dynamic>();

		jsonDados['id'] = this.id ?? 0;
		jsonDados['idBancoContaCaixa'] = this.idBancoContaCaixa ?? 0;
		jsonDados['mesAno'] = Biblioteca.removerMascara(this.mesAno);
		jsonDados['mes'] = this.mes;
		jsonDados['ano'] = this.ano;
		jsonDados['dataMovimento'] = this.dataMovimento != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.dataMovimento) : null;
		jsonDados['dataBalancete'] = this.dataBalancete != null ? DateFormat('yyyy-MM-ddT00:00:00').format(this.dataBalancete) : null;
		jsonDados['historico'] = this.historico;
		jsonDados['documento'] = this.documento;
		jsonDados['valor'] = this.valor;
		jsonDados['conciliado'] = setConciliado(this.conciliado);
		jsonDados['observacao'] = this.observacao;
		jsonDados['bancoContaCaixa'] = this.bancoContaCaixa == null ? null : this.bancoContaCaixa.toJson;
	
		return jsonDados;
	}
	
    getConciliado(String conciliado) {
    	switch (conciliado) {
    		case 'S':
    			return 'Sim';
    			break;
    		case 'N':
    			return 'Não';
    			break;
    		default:
    			return null;
    		}
    	}

    setConciliado(String conciliado) {
    	switch (conciliado) {
    		case 'Sim':
    			return 'S';
    			break;
    		case 'Não':
    			return 'N';
    			break;
    		default:
    			return null;
    		}
    	}


	String objetoEncodeJson(FinExtratoContaBanco objeto) {
	  final jsonDados = objeto.toJson;
	  return json.encode(jsonDados);
	}
	
}

File: projetos-fenix-para-estudo/t2ti-erp-fenix-cadastros-completo/fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/model/financeiro/FinExtratoContaBanco.java
/*******************************************************************************
Title: T2Ti ERP Fenix                                                                
Description: Model relacionado à tabela [FIN_EXTRATO_CONTA_BANCO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2020 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.model.financeiro;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.math.BigDecimal;
import java.util.Date;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.t2ti.fenix.model.cadastros.BancoContaCaixa;

import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;

@Entity
@Table(name="FIN_EXTRATO_CONTA_BANCO")
@NamedQuery(name="FinExtratoContaBanco.findAll", query="SELECT t FROM FinExtratoContaBanco t")
public class FinExtratoContaBanco implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="MES_ANO")
	private String mesAno;

    @Column(name="MES")
	private String mes;

    @Column(name="ANO")
	private String ano;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_MOVIMENTO")
	private Date dataMovimento;

    @Temporal(TemporalType.DATE)
	@Column(name="DATA_BALANCETE")
	private Date dataBalancete;

    @Column(name="HISTORICO")
	private String historico;

    @Column(name="DOCUMENTO")
	private String documento;

    @Column(name="VALOR")
	private BigDecimal valor;

    @Column(name="CONCILIADO")
	private String conciliado;

    @Column(name="OBSERVACAO")
	private String observacao;

    @ManyToOne
	@JoinColumn(name="ID_BANCO_CONTA_CAIXA")
    private BancoContaCaixa bancoContaCaixa;

	public FinExtratoContaBanco() {
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public String getMesAno() {
		return this.mesAno;
	}

	public void setMesAno(String mesAno) {
		this.mesAno = mesAno;
	}

    public String getMes() {
		return this.mes;
	}

	public void setMes(String mes) {
		this.mes = mes;
	}

    public String getAno() {
		return this.ano;
	}

	public void setAno(String ano) {
		this.ano = ano;
	}

    public Date getDataMovimento() {
		return this.dataMovimento;
	}

	public void setDataMovimento(Date dataMovimento) {
		this.dataMovimento = dataMovimento;
	}

    public Date getDataBalancete() {
		return this.dataBalancete;
	}

	public void setDataBalancete(Date dataBalancete) {
		this.dataBalancete = dataBalancete;
	}

    public String getHistorico() {
		return this.historico;
	}

	public void setHistorico(String historico) {
		this.historico = historico;
	}

    public String getDocumento() {
		return this.documento;
	}

	public void setDocumento(String documento) {
		this.documento = documento;
	}

    public BigDecimal getValor() {
		return this.valor;
	}

	public void setValor(BigDecimal valor) {
		this.valor = valor;
	}

    public String getConciliado() {
		return this.conciliado;
	}

	public void setConciliado(String conciliado) {
		this.conciliado = conciliado;
	}

    public String getObservacao() {
		return this.observacao;
	}

	public void setObservacao(String observacao) {
		this.observacao = observacao;
	}

    public BancoContaCaixa getBancoContaCaixa() {
		return this.bancoContaCaixa;
	}

	public void setBancoContaCaixa(BancoContaCaixa bancoContaCaixa) {
		this.bancoContaCaixa = bancoContaCaixa;
	}

		
}